Description: Keep default grub-dpkg support
This module is no longer needed in newer releases.
Revert its removal as a default module.

Author: James Falcon <james.falcon@canonical.com>
Last-Update: 2024-11-11
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/cloudinit/config/cc_grub_dpkg.py
+++ b/cloudinit/config/cc_grub_dpkg.py
@@ -22,7 +22,7 @@ meta: MetaSchema = {
     "id": "cc_grub_dpkg",
     "distros": ["ubuntu", "debian"],
     "frequency": PER_INSTANCE,
-    "activate_by_schema_keys": ["grub_dpkg", "grub-dpkg"],
+    "activate_by_schema_keys": [],
 }
 
 LOG = logging.getLogger(__name__)
@@ -113,7 +113,7 @@ def handle(name: str, cfg: Config, cloud
     if not mycfg:
         mycfg = {}
 
-    enabled = mycfg.get("enabled", False)
+    enabled = mycfg.get("enabled", True)
     if util.is_false(enabled):
         LOG.debug("%s disabled by config grub_dpkg/enabled=%s", name, enabled)
         return
--- a/cloudinit/config/schemas/schema-cloud-config-v1.json
+++ b/cloudinit/config/schemas/schema-cloud-config-v1.json
@@ -1655,8 +1655,8 @@
           "properties": {
             "enabled": {
               "type": "boolean",
-              "default": false,
-              "description": "Whether to configure which device is used as the target for grub installation. Default: ``false``."
+              "default": true,
+              "description": "Whether to configure which device is used as the target for grub installation. Default: ``true``."
             },
             "grub-pc/install_devices": {
               "type": "string",
--- a/tests/unittests/config/test_cc_grub_dpkg.py
+++ b/tests/unittests/config/test_cc_grub_dpkg.py
@@ -248,7 +248,7 @@ class TestHandle:
         """Test setting of correct debconf database entries"""
         m_is_efi_booted.return_value = is_uefi
         m_fetch_idevs.return_value = fetch_idevs_output
-        cfg = {"grub_dpkg": {"enabled": True}}
+        cfg = {"grub_dpkg": {}}
         if cfg_idevs is not None:
             cfg["grub_dpkg"]["grub-pc/install_devices"] = cfg_idevs
         if cfg_idevs_empty is not None:
